x-pack/plugins/observability_ai_assistant/public/hooks/use_json_editor_model.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { useMemo } from 'react';
+import { monaco } from '@kbn/monaco';
+import { FunctionDefinition } from '../../common/types';
+
+const { editor, languages, Uri } = monaco;
+
+const SCHEMA_URI = 'http://elastic.co/foo.json';
+const modelUri = Uri.parse(SCHEMA_URI);
+
+export const useJsonEditorModel = (functionDefinition?: FunctionDefinition) => {
+  return useMemo(() => {
+    if (!functionDefinition) {
+      return {};
+    }
+
+    const schema = { ...functionDefinition.options.parameters };
+
+    const initialJsonString = functionDefinition.options.parameters.properties
+      ? Object.keys(functionDefinition.options.parameters.properties).reduce(
```

Comment:
This currently assumes properties is a flat object. Not quite sure if that is the case always. Cc @dgieselaar

Reply:
It's not. You should be able to tell from the schema whether to create an object or an empty string. I think it's a nice addition but if it's too complicated to get a correct initial value we should just stick to an empty object (so `{}`).

Reply:
I think it's cool addition that makes the experience a bit nicer. Will pick this up in a new PR.